holmes/plugins/toolsets/bash/bash_toolset.py

```diff
@@ -0,0 +1,303 @@
+        super().__init__(
+            name="kubectl_run_image",
+            description=(
+                "Executes `kubectl run <name> --image=<image> ... -- <command>` return the result"
+            ),
+            parameters={
+                "image": ToolParameter(
+                    description="The image to run",
+                    type="string",
+                    required=True,
+                ),
+                "command": ToolParameter(
+                    description="The command to execute on the deployed pod",
+                    type="string",
+                    required=True,
+                ),
+                "namespace": ToolParameter(
+                    description="The namespace in which to deploy the temporary pod",
+                    type="string",
+                    required=False,
+                ),
+                "timeout": ToolParameter(
+                    description=(
+                        "Optional timeout in seconds for the command execution. "
+                        "Defaults to 60s."
+                    ),
+                    type="integer",
+                    required=False,
+                ),
+            },
+            toolset=toolset,
+        )
+
+    def _build_kubectl_command(self, params: dict, pod_name: str) -> str:
+        namespace = params.get("namespace", "default")
+        image = get_param_or_raise(params, "image")
+        command_str = get_param_or_raise(params, "command")
+        return f"kubectl run {pod_name} --image={image} --namespace={namespace} --rm --attach --restart=Never -i --tty -- {command_str}"
+
+    def _invoke(self, params: Dict[str, Any]) -> StructuredToolResult:
+        timeout = params.get("timeout", 60)
+
+        image = get_param_or_raise(params, "image")
+        command_str = get_param_or_raise(params, "command")
+
+        namespace = params.get("namespace")
+
+        if namespace and not re.match(SAFE_NAMESPACE_PATTERN, namespace):
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data=f"Error: The namespace is invalid. Valid namespaces must match the following regexp: {SAFE_NAMESPACE_PATTERN}",
+                params=params,
+            )
+
+        validate_image_and_commands(
+            image=image, container_command=command_str, config=self.toolset.config
+        )
+
+        pod_name = (
+            "holmesgpt-debug-pod-"
+            + "".join(random.choices(string.ascii_letters, k=8)).lower()
+        )
+        full_kubectl_command = self._build_kubectl_command(params, pod_name)
+        try:
+            process = subprocess.run(
+                full_kubectl_command,
+                shell=True,
+                executable="/bin/bash",
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+                timeout=timeout,
+                check=False,
+            )
+
+            result_data = (
```

Comment:
Thinking here loud. 
Don't we want to seperate the stdout & stderr from each other and take only the relevant output?
Another thing. we added support for no_data in status. In the current way we adding the full kubectl command will be confusing. Why we adding it there? can we also make sure to return no_data in relevant cases?